tests/test_bit_manip.cc

```diff
@@ -76,4 +76,28 @@ TEMPLATE_TEST_CASE("Access MSB", "[bit_manip]",
     REQUIRE(gul14::bit_test(x, msb - 1) == false);
 }
 
+TEST_CASE("is_little_endian(), is_big_endian()", "[bit_manip]")
+{
+    REQUIRE(gul14::is_little_endian() != gul14::is_big_endian());
+
+    const std::uint32_t val = 0x01'02'03'04;
+    const void* ptr = &val;
+    const char* chars = static_cast<const char*>(ptr);
```

Comment:
Imho this should be a `reinterpret_cast` to make the intention more clear.

Reply:
No. A `reinterpret_cast` entails undefined behavior, a `static_cast` is blessed by the standard.

Reply:
> `static_cast` conversion
>
> Converts between types using a combination of implicit and user-defined conversions.

Furthermore allowed ('blessed') by the standard is only `static_cast` to and from `void*`, which does look ugly

`const char* chars = static_cast<const char*>(static_cast<void*>(ptr));`

But especially you want **no** conversion to happen at all:

> `reinterpret_cast` conversion
>
> Converts between types by reinterpreting the underlying bit pattern.

And **that** is what you want, right?

The whole code is anyhow UB (or is it implementation defined behavior, which would be ok), and that makes it more explicit that we want to do something fishy here.

Reply:
> And that is what you want, right?

Agreed, and changed to `reinterpret_cast`.

> The whole code is anyhow UB (or is it implementation defined behavior, which would be ok), and that makes it more explicit that we want to do something fishy here.

No, it is not UB, but implementation defined for the special case of casting to some kind of `char*` or `std::byte*`. That is what I initially missed – I thought that any `reinterpret_cast` was automatically UB, which is not true.